src/middlewared/middlewared/plugins/chart_releases_linux/upgrade.py

```diff
@@ -0,0 +1,119 @@
+import errno
+import os
+import shutil
+import tempfile
+import yaml
+
+from pkg_resources import parse_version
+
+from middlewared.schema import Dict, Str
+from middlewared.service import accepts, CallError, Service, ValidationErrors
+
+from .schema import clean_values_for_upgrade
+from .utils import get_namespace, run
+
+
+class ChartReleaseService(Service):
+
+    class Config:
+        namespace = 'chart.release'
+
+    @accepts(
+        Str('release_name'),
+        Dict(
+            'upgrade_options',
+            Dict('values', additional_attrs=True),
+            Str('item_version', required=True),
+        )
+    )
```

Comment:
Seems like this should be a job?

Reply:
We are not actually waiting for the upgrade to complete. We just tell k8s that we want the workloads to be in this state and it performs the upgrade itself in the background.
I have been thinking of using events to notify the user wrt upgrades / creation of chart releases.

Reply:
Still might make sense to make it a job, just t o be consistent between the CRUD methods